src/init.cpp

```diff
@@ -1631,7 +1631,7 @@ bool AppInitMain(InitInterfaces& interfaces)
 
     // ********************************************************* Step 11: import blocks
 
-    if (!CheckDiskSpace() && !CheckDiskSpace(0, true))
+    if (!CheckDiskSpace() || !CheckDiskSpace(/* additional_bytes */ 0, /* blocks_dir */ true))
```

Comment:
When you are changing this anyway: please provide explicit arguments to the first call too, instead of relying on default arguments